src/account/operations/transfer/high_level/sweep_outputs.rs

```diff
@@ -0,0 +1,412 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{borrow::Borrow, collections::HashSet, pin::Pin, str::FromStr};
+
+use futures::{Future, FutureExt};
+use iota_client::{
+    api::ClientMessageBuilder,
+    bee_message::{
+        address::{Address, AliasAddress, NftAddress},
+        output::{
+            dto::OutputDto,
+            unlock_condition::{
+                dto::{AddressUnlockConditionDto, UnlockConditionDto},
+                AddressUnlockCondition,
+            },
+            AliasId, AliasOutput, AliasOutputBuilder, FoundryId, NftId, NftOutput, Output, OutputId, TokenScheme,
+            OUTPUT_COUNT_MAX,
+        },
+        payload::transaction::TransactionId,
+    },
+    bee_rest_api::types::responses::OutputResponse,
+    node_api::indexer::query_parameters::QueryParameter,
+};
+
+use crate::{
+    account::{
+        handle::AccountHandle,
+        types::{address::AddressWrapper, AccountAddress, OutputData},
+        AddressGenerationOptions, RemainderValueStrategy, TransferOptions,
+    },
+    Error,
+};
+
+#[allow(dead_code)]
+impl AccountHandle {
+    pub(crate) async fn get_sweep_remainder_address(
+        &self,
+        options: &Option<TransferOptions>,
+    ) -> crate::Result<AddressWrapper> {
+        let gen_addr = async {
+            let result: crate::Result<AccountAddress> = Ok(self
+                .generate_addresses(
+                    1,
+                    Some(AddressGenerationOptions {
+                        internal: true,
+                        ..Default::default()
+                    }),
+                )
+                .await?
+                .first()
+                .ok_or_else(|| crate::Error::BurningFailed("Couldn't generate an address".to_string()))?
+                .clone());
+
+            result
+        };
```

Comment:
Maybe this should be an own `pub(crate) async fn generate_remainder_address()` method